demo-projects/meetup/site/lib/apollo.js

```diff
@@ -0,0 +1,62 @@
+// Derived from https://github.com/zeit/next.js/tree/2789e7e0c25c72fbf6be3ef070e530186af14430/examples/with-apollo
+import React from 'react';
+import { ApolloProvider } from '@apollo/react-hooks';
+import createApolloClient from './apolloClient';
+
+// On the client, we store the Apollo Client in the following variable.
+// This prevents the client from reinitializing between page transitions.
+let globalApolloClient = null;
+
+/**
+ * Always creates a new apollo client on the server
+ * Creates or reuses apollo client in the browser.
+ * @param  {NormalizedCacheObject} initialState
+ * @param  {NextPageContext} ctx
+ */
+export const initApolloClient = (initialState, ctx) => {
+  // Make sure to create a new client for every server-side request so that data
+  // isn't shared between connections (which would be bad)
+  if (typeof window === 'undefined') {
+    return createApolloClient(initialState, ctx);
+  }
+
+  // Reuse client on the client-side
+  if (!globalApolloClient) {
+    globalApolloClient = createApolloClient(initialState, ctx);
+  }
+
+  return globalApolloClient;
+};
+
+/**
+ * Creates a withApollo HOC
+ * that provides the apolloContext
+ * to a next.js Page or AppTree.
+ * @returns {(PageComponent: ReactNode) => ReactNode}
+ */
+export const withApollo = PageComponent => {
```

Comment:
I don't think we should use `withApollo` anymore. Maybe the `ApolloProvider` should just wrap the main `_app.js` https://github.com/keystonejs/keystone/pull/3229/files#diff-60368d2a025f753c4704333759a8689bR14 like the other providers? 

See:
https://github.com/vercel/next.js/blob/canary/examples/with-apollo/pages/_app.js

I think we had a `withApollo` initially because there was a lot more set-up involved.